xoto3/dynamodb/write_versioned/lazy_batch_gets.py

```diff
@@ -0,0 +1,56 @@
+from typing import Collection, Mapping, Tuple
+
+from .errors import ItemNotFetchedException
+from .modify import presume
+from .prepare import add_item_to_base_request, parse_batch_get_request, prepare_clean_transaction
+from .types import BatchGetItem, ItemKey, TransactionBuilder, VersionedTransaction
+
+
+def lazy_batch_getting_transaction_builder(
+    transaction_builder: TransactionBuilder,
+    item_keys_by_table_name: Mapping[str, Collection[ItemKey]],
+    batch_get_item: BatchGetItem,
+) -> Tuple[VersionedTransaction, VersionedTransaction]:
+    """If your transaction attempts to get/require an item that was not
+    prefetched, we will stop, fetch it, and then retry your
+    transaction from the beginning.
+
+    Remember to make your transaction a pure function, or at least make
```

Comment:
I actually ended up not being able to figure out how to make my most recent transaction a pure function. I should show you that, and maybe you can help me refactor it into one.